Lib/test/test_io.py

```diff
@@ -407,10 +407,12 @@ def large_file_ops(self, f):
     def test_invalid_operations(self):
         # Try writing on a file opened in read mode and vice-versa.
         exc = self.UnsupportedOperation
-        for mode in ("w", "wb"):
-            with self.open(os_helper.TESTFN, mode) as fp:
-                self.assertRaises(exc, fp.read)
-                self.assertRaises(exc, fp.readline)
+        with self.open(os_helper.TESTFN, "w", encoding="ascii") as fp:
```

Comment:
I would prefer to replace all "ascii" with "utf-8". Come on, ASCII is now legacy!